src/plugins/info/info_impl.cpp

```diff
@@ -167,63 +157,63 @@ void InfoImpl::process_flight_information(const mavlink_message_t& message)
     _flight_information_received = true;
 }
 
-void InfoImpl::translate_binary_to_str(
-    uint8_t* binary, unsigned binary_len, char* str, unsigned str_len)
+std::string InfoImpl::translate_binary_to_str(uint8_t* binary, unsigned binary_len)
 {
+    std::string str(binary_len * 2, '0');
+
     for (unsigned i = 0; i < binary_len; ++i) {
         // One hex number occupies 2 chars.
         // The binary is in little endian, therefore we need to swap the bytes for us to read.
-        snprintf(&str[i * 2], str_len - i * 2, "%02x", binary[binary_len - 1 - i]);
+        snprintf(&str[i * 2], str.length() - i * 2, "%02x", binary[binary_len - 1 - i]);
     }
+
+    return str;
 }
 
 std::pair<Info::Result, Info::Identification> InfoImpl::get_identification() const
 {
     std::lock_guard<std::mutex> lock(_mutex);
     return std::make_pair<>(
-        (_information_received ? Info::Result::SUCCESS :
-                                 Info::Result::INFORMATION_NOT_RECEIVED_YET),
+        (_information_received ? Info::Result::Success : Info::Result::InformationNotReceivedYet),
         _identification);
 }
 
 std::pair<Info::Result, Info::Version> InfoImpl::get_version() const
 {
     std::lock_guard<std::mutex> lock(_mutex);
     return std::make_pair<>(
-        (_information_received ? Info::Result::SUCCESS :
-                                 Info::Result::INFORMATION_NOT_RECEIVED_YET),
+        (_information_received ? Info::Result::Success : Info::Result::InformationNotReceivedYet),
         _version);
 }
 
 std::pair<Info::Result, Info::Product> InfoImpl::get_product() const
 {
     std::lock_guard<std::mutex> lock(_mutex);
     return std::make_pair<>(
-        (_information_received ? Info::Result::SUCCESS :
-                                 Info::Result::INFORMATION_NOT_RECEIVED_YET),
+        (_information_received ? Info::Result::Success : Info::Result::InformationNotReceivedYet),
         _product);
 }
 
 std::pair<Info::Result, Info::FlightInfo> InfoImpl::get_flight_information() const
 {
     std::lock_guard<std::mutex> lock(_mutex);
     return std::make_pair<>(
-        (_flight_information_received ? Info::Result::SUCCESS :
-                                        Info::Result::INFORMATION_NOT_RECEIVED_YET),
+        (_flight_information_received ? Info::Result::Success :
+                                        Info::Result::InformationNotReceivedYet),
         _flight_info);
 }
 
-const char* InfoImpl::vendor_id_str(uint16_t vendor_id)
+const std::string InfoImpl::vendor_id_str(uint16_t vendor_id)
 {
     switch (vendor_id) {
         case 0x26ac:
-            return "Yuneec";
+            return "3D Robotics Inc.";
```

Comment:
@julianoes: worth your attention, too. Note that I did not change "H520" below (line 219)